float64 x_rel
float64 y_rel
